tests/components/nextcloud/test_config_flow.py

```diff
@@ -0,0 +1,174 @@
+"""Tests for the Nextcloud config flow."""
+from unittest.mock import Mock, patch
+
+from nextcloudmonitor import NextcloudMonitorError
+import pytest
+
+from homeassistant.components.nextcloud import DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_user_create_entry(
+    hass: HomeAssistant, mock_nextcloud_monitor: Mock
+) -> None:
+    """Test that the user step works."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == SOURCE_USER
```

Comment:
The "step" is not the same as the "source", so `SOURCE_USER` isn't really valid here.
```suggestion
    assert result["step_id"] == "user"
```